.github/workflows/unit-tests.yml

```diff
@@ -120,8 +120,24 @@ jobs:
           cp -R /tmp/MNIST .
 
       - name: Run Tests
-        run: |
-          SKIP_DISTRIB_TESTS=${{ matrix.skip-distrib-tests }} bash tests/run_cpu_tests.sh
+        uses: nick-fields/retry@v3
+        with:
+          max_attempts: 5
+          timeout_minutes: 20
+          shell: bash
+          command: SKIP_DISTRIB_TESTS=${{ matrix.skip-distrib-tests }} bash tests/run_cpu_tests.sh
+          on_retry_command: |
+            echo sending kill signal until process group no longer exists...
+            if [ -f .ignite_testing.pid ] && [ -s .ignite_testing.pid ]; then
+              tests_pid=$(cat .ignite_testing.pid)
+              kill -0 -- -$tests_pid && kill -INT -- -$tests_pid && sleep 5
+              set -x
+              while kill -0 -- -$tests_pid > /dev/null 2>&1; do kill -9 -- -$tests_pid; sleep 3; done
+            else
+              echo "File .ignite_testing.pid does not exist or is empty"
+            fi
```

Comment:
If there is a failure in tests on windows, will this all work ?
Also, why we need to do all this for unit-tests.yml and not for hvd or gpu tests?

Reply:
I've moved the functionality into python to support windows a bit better and make the approach more robust but I don't know anything about process management/signals on windows so hopefully the minimal attempt for that OS is enough.